go/vt/wrangler/shard.go

```diff
@@ -114,6 +114,10 @@ func (wr *Wrangler) SetShardTabletControl(ctx context.Context, keyspace, shard s
 func (wr *Wrangler) DeleteShard(ctx context.Context, keyspace, shard string, recursive bool) error {
 	shardInfo, err := wr.ts.GetShard(ctx, keyspace, shard)
 	if err != nil {
+		if err == topo.ErrNoNode {
```

Comment:
We don't know what cells to look at... no shard record means no Cells list.

We could use all topo server cells though, and if the ShardReplication exists, use it to find tablets (as we do already). That would mean rewriting this function in a different way though.

Is it worth it?